src/providers/trusted_service/context/mod.rs

```diff
@@ -171,11 +196,14 @@ impl Context {
         Ok(ctx)
     }
 
+    // Serialize and send a request and deserialize the response back to
+    // the caller. The caller is responsible for explicitly declaring
+    // the response type if its contents are of interest.
     fn send_request<T: Message + Default>(
         &self,
         req: &(impl Message + GetOpcode),
     ) -> Result<T, PsaError> {
-        let _mutex_guard = self.call_mutex.try_lock().expect("Call mutex poisoned");
+        let _mutex_guard = self.call_mutex.lock().expect("Call mutex poisoned");
```

Comment:
Rookie mistake 👎

Reply:
Don't we want the thread to wait at this point that the mutex is free?

Reply:
We do, but `try_lock` just returns an error if the lock is already taken. `lock` returns an error when it's poisoned

Reply:
ahh yes right `lock` is the right one